hack/cluster-clean.sh

```diff
@@ -47,29 +43,29 @@ function remove_finalizers() {
         local arr=($p)
         local name="${arr[0]}"
         local ns="${arr[1]}"
-        _kubectl patch vmsnapshots $name -n $ns --type=json -p '[{ "op": "remove", "path": "/metadata/finalizers" }]'
+        patch_remove_finalizers -n $ns vmsnapshots $name
     done
```

Comment:
idea for following PR

snapshots=$(kubectl get vmsnapshots --all-namespaces -o=custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace,FINALIZERS:.metadata.finalizers --no-headers | grep vmsnapshot-protection)

and then use some awk to operate line by line on the output
(since its a function the function will be need to be exported first using `typeset`
same for all the rest in this function.

Note that the command above shows also lines without finalizers, so the finalizers field can be remove from the command, (and we need to make sure the patch doesnt fail in case the finalizers isn't found, else its a bug that exists now already)

Reply:
I'm not sure I understand your suggestion, but I certainly notice the repeated code of these `while` loops. I would love to simplify it in a follow-up, however I'd like to first understand why we care about which finalizers we remove. My gut feeling is that these `grep -e` for specific finalizers are not helping us.